### PR TITLE
Let Users restrict reviews to customer with purchased item

### PR DESCRIPTION
**Pull request by smohamadabedy**
I think this is good to let user restricts  reviews  and only customers with purchased  item can add reviw. 
admin -> Setting and catalog -> product are updated.

### CODE REVIEW

**Comment by TheCartpenter** [comment]
I never really liked these sorts of features because whenever customers wants to purchase their goods, and they never tried them before, they'd have to send these reviews before officially buying the items when the philosophy is rather about: Try before you buy .., not: rate before you try.

File: upload/catalog/controller/product/product.php [hunk 1]
```diff
@@ -416,6 +416,13 @@ public function index(): void {
 				$data['review_guest'] = false;
 			}
 
+			if ($this->config->get('config_review_on_purchase') && $this->customer->isLogged()) {
+				$data['review_on_purchase'] = $this->model_catalog_product->cutomerHasPurchase($this->customer->getId(),$product_id) > 0 ? true : false;
+			}else{
+				$data['review_on_purchase'] = true;
+			}
+			var_dump($this->config->get('config_review_on_purchase') ,$this->model_catalog_product->cutomerHasPurchase($this->customer->getId(),$product_id) > 0,$data['review_guest'] );
```
Comment (TheCartpenter) [inline upload/catalog/controller/product/product.php:424]: Is this needed for specific reasons?

Comment (smohamadabedy) [reply upload/catalog/controller/product/product.php:424]: no actually was for testing I'll delete it.

**Comment by smohamadabedy** [comment]
> I never really liked these sorts of features because whenever customers wants to purchase their goods, and they never tried them before, they'd have to send these reviews before officially buying the items when the philosophy is rather about: Try before you buy .., not: rate before you try.

I'll saw this in feature release To do section, And actually I really like it to be in opencart as I close review section for my opencart customers most of the times.

**Comment by TheCartpenter** [comment]
Yes, once the item has been purchased. Not before would be a great fit to the core.

**Comment by danielkerr** [comment]
https://github.com/opencart/opencart/issues/9417

**Pull request closed without merging**